File: 프로그래머스/1/12901. 2016년/2016년.java
import java.util.*;

class Solution {
    public String solution(int a, int b) {
        int year = 2016;
        Calendar c = Calendar.getInstance();
        c.set(year, a - 1, b);
        int d = c.get(Calendar.DAY_OF_WEEK);
        String answer = "";
        switch (d){
            case 1:
                answer = "SUN";
                break;
            case 2:
                answer = "MON";
                break;
            case 3:
                answer = "TUE";
                break;
            case 4:
                answer = "WED";
                break;
            case 5:
                answer = "THU";
                break;
            case 6:
                answer = "FRI";
                break;
            case 7:
                answer = "SAT";
        }
        return answer;
    }
}

File: 프로그래머스/1/12901. 2016년/2016년.swift
import Foundation

func solution(_ a:Int, _ b:Int) -> String {
    var result = ""
    let dateString = "2016-\(a)-\(b)"
    let dateFormatter = DateFormatter()
    dateFormatter.dateFormat = "yyyy-MM-dd"
    guard let date = dateFormatter.date(from: dateString) else {
        return result
    }
    let calendar = Calendar.current
    let weekDay = calendar.component(.weekday, from: date)
    
    switch weekDay {
    case 1:
        result = "SUN"
    case 2:
        result = "MON"
    case 3:
        result = "TUE"
    case 4:
        result = "WED"
    case 5:
        result = "THU"
    case 6:
        result = "FRI"
    case 7:
        result = "SAT"
    default: 
        result = "no"
    }
    return result
}